experimenter/experimenter/changelog/views.py

```diff
@@ -1 +1,31 @@
-# Create your views here.
+from django.shortcuts import get_object_or_404
+from django.views.generic import ListView
+
+from experimenter.experiments.models import NimbusChangeLog, NimbusExperiment
+
+
+class NimbusChangeLogsView(ListView):
+    model = NimbusChangeLog
+    template_name = "changelog/test.html"
+    context_object_name = "experiment"
+
+    def get(self, request, *args, **kwargs):
+        self.object = self.get_object()
+        return super().get(request, *args, **kwargs)
+
+    def get_object(self, queryset=None):
+        slug = self.kwargs.get("slug")
+        experiment = get_object_or_404(NimbusExperiment, slug=slug)
+        return experiment
+
+    def get_context_data(self, **kwargs):
+        slug = self.kwargs.get("slug")
+        context = super().get_context_data(**kwargs)
+        experiment = self.object
+
+        # Retrieve the changelogs for the experiment
+        changelogs = experiment.changes.all()
```

Comment:
I would make this `changelogs = list(experiment.changes.all())` because `.all()` returns a QuerySet and iterating over it does a DB query. If the template ever ended up iterating over `changelogs` twice, it would do two DB queries.

Reply:
Oh, I didn't know that. I'll update the PR accordingly , thanks :)